meta-iotconnect/recipes-apps/temp/files/0002-app-config.patch

```diff
@@ -0,0 +1,59 @@
+From 982095bca4f90688c3387f6aae602ccddc85b3e2 Mon Sep 17 00:00:00 2001
+From: "P.Young" <pyoung@witekio.com>
+Date: Tue, 18 Apr 2023 09:28:07 +0100
+Subject: [PATCH] changed preprocessors to allow for compiler argument
+ over-ride
+
+---
+ samples/basic-sample/config/app_config.h | 21 ++++++++++++++++++---
+ 1 file changed, 18 insertions(+), 3 deletions(-)
+
+diff --git a/samples/basic-sample/config/app_config.h b/samples/basic-sample/config/app_config.h
+index b70c45f..fc3bdd5 100644
+--- a/samples/basic-sample/config/app_config.h
++++ b/samples/basic-sample/config/app_config.h
+@@ -7,23 +7,38 @@
+ 
+ #include "iotconnect.h"
+ 
+-#define IOTCONNECT_CPID "your-cpid"
+-#define IOTCONNECT_ENV  "your-enviroment"
++#ifndef IOTCONNECT_CPID
++#error "IOTCONNECT_CPID not defined, please define in compiler args/enviroment variables or similar."
```

Comment:
ENV and CPID are in the Key Vault at the IoTConnect WEB UI and some users may not look there as it is not really intuitive. Some comments on where to locate these values would be useful since you are adding other useful comments to the config file. Up to you though.

Reply:
patch removed with a bespoke app-config.h that is added in the example layer: hopefully fixed in commit 6aec1859de38af5d854d509bdbf327233c7d0022

Reply:
The point that I was trying to make is that I have hoped that you would adapt and improve app_config.h in the original repo.

I am dropping this issue and acceptring your resotuion. Feel free to revert back to the original patch approach, if youi feel that it is better than adding the app_config.h to bbappend.